vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/VoiceMessagePlaybackTracker.kt

```diff
@@ -26,8 +26,17 @@ class VoiceMessagePlaybackTracker @Inject constructor() {
 
     private val mainHandler = Handler(Looper.getMainLooper())
     private val listeners = mutableMapOf<String, Listener>()
+    private val activityListeners = mutableMapOf<String, ActivityListener>()
     private val states = mutableMapOf<String, Listener.State>()
 
+    fun trackActivity(key: String, listener: ActivityListener) {
+        activityListeners[key] = listener
```

Comment:
What is the advantage to have a `key` here? I see that it's easier to remove the listener since the caller does not have to pass it.
Maybe log a warning if `activityListeners[key]` already exist?

Reply:
it was mainly for being able to remove the listener, happy to switch to something instance based

Reply:
https://github.com/vector-im/element-android/pull/4557/commits/28af0a637fb5d0fa1b77ba741c306cfeb9f7bcea also renames the parameter